Mathlib/Data/Finset/Lattice.lean

```diff
@@ -568,35 +568,27 @@ theorem comp_sup_eq_sup_comp_of_is_total [SemilatticeSup β] [OrderBot β] (g :
 @[simp]
 protected theorem le_sup_iff (ha : ⊥ < a) : a ≤ s.sup f ↔ ∃ b ∈ s, a ≤ f b := by
   apply Iff.intro
-  · intro h'
-    induction s using Finset.cons_induction with
-    | h₁ => exact False.elim (not_le_of_lt ha h')
-    | @h₂ c s hc ih =>
-      rw [sup_cons, _root_.le_sup_iff] at h'
-      cases h' with
-      | inl h => exact ⟨c, mem_cons.2 (Or.inl rfl), h⟩
-      | inr h =>
-        rcases ih h with ⟨b, hb, hle⟩
-        exact ⟨b, mem_cons.2 (Or.inr hb), hle⟩
-  · rintro ⟨b, hb, hle⟩
-    exact le_trans hle (le_sup hb)
+  · induction s using cons_induction with
+    | empty => exact (absurd · (not_le_of_lt ha))
+    | @cons c t hc ih =>
```

Comment:
The `@` here surprises me a bit. Are you allowed to write `cons {c t} hc ih` or similar instead?

Reply:
Hmm no Lean doesn't seem to like that here. I think I've seen it work with `fun` though